### ISSUE TITLE
Query about the range of inputs valid for xfastpowf_u3500?

### ISSUE THREAD

**Issue by sheredom** [shibatch/sleef#273]
I've been evaluating using `xfastpowf_u3500` for our low precision pow calculations as part of my work at Unity on the Burst compiler, and I'm getting some finite range inputs producing unexpected 0's. I was wondering if this is expected? I couldn't find anywhere in the documentation whether there was a subset of the floating-point range valid for `xfastpowf_u3500`.

For instance, comparing against a reference:

```
1) Failed : Burst.Compiler.IL.Tests.TestUnityMathPrecision.TestPowLow(-100.0f,-77.9f)
  Expected: NaN != Result: 0, ULPs: 2147483647
  Expected: True
  But was:  False
```

I guess that the 0 is being produced because the `exp2` in `exp2(log2(x) * y)` would produce a 0 value (as the input would be very small). The non-fast version produces a NaN because canonically the result of this calculation would be an imaginary number I believe.

If this is simply a case where `xfastpowf_u3500` is expected to not generate the correct NaNs that is fine, I just wanted to query whether that was expected or not?

Any guidance you could provide would be very helpful!

**Comment by shibatch** [comment]
Hello,

I added fastpowf function in response to the request made by Unity.
Since @xoofx said that the computation speed has the highest priority, I removed all the checks for special value handling. Therefore, it is kind of expected that the function returns strange values on some specific inputs.
I would rather know if this is okay for you or not.
If not, I will fix the function to reflect your needs.
